Src/AutoFixtureUnitTest/Kernel/TerminatingWithPathSpecimenBuilderTest.cs

```diff
@@ -162,35 +159,62 @@ public void SpecimenRequestsWillInitiallyBeEmpty()
-            var sut = new TerminatingWithPathSpecimenBuilder(tracer);
-            // Exercise system
-            requestEvents.ForEach(tracer.RaiseSpecimenRequested);
-            // Verify outcome
-            Assert.True(specimens.SequenceEqual(sut.SpecimenRequests));
-            // Teardown
+            // arrange
+            var requests = new[] { new object(), new object(), new object() };
+            var requestQueue = new Queue<object>(requests);
+            var firstRequest = requestQueue.Dequeue();
+
+            object[] capturedRequests = null;
+            TerminatingWithPathSpecimenBuilder sut = null;
+            
+            var builder = new DelegatingSpecimenBuilder
+            {
+                OnCreate = (r, c) =>
+                {
+                    if(requestQueue.Count > 0) 
+                        return c.Resolve(requestQueue.Dequeue());
+                    
+                    capturedRequests = sut.SpecimenRequests.ToArray();
+                    return new object();
+                }
+            };
+            
+            sut = new TerminatingWithPathSpecimenBuilder(builder);
+            // Cause sut to be executed recursively for multiple times.
+            var context = new SpecimenContext(sut);
+            
+            //act
+            sut.Create(firstRequest, context);
+            
+            //assert
+            Assert.NotNull(capturedRequests);
+            Assert.Equal(requests, capturedRequests);
         }
 
         [Fact]
         public void SpecimenRequestsAreEmptyWhenAllThatWereRequestedAreCreated()
         {
-            // Fixture setup
-            var tracer = new DelegatingTracingBuilder();
-            var specimens = new[] { new object(), new object(), new object() };
-            var requestEvents = specimens.Select(
-                (o, i) => new RequestTraceEventArgs(o, i)).ToList();
-            var createdEvents = specimens.Reverse().Select(
-                (o, i) => new SpecimenCreatedEventArgs(o, null, i)).ToList();
-            var sut = new TerminatingWithPathSpecimenBuilder(tracer);
-            // Exercise system
-            requestEvents.ForEach(tracer.RaiseSpecimenRequested);
-            createdEvents.ForEach(tracer.RaiseSpecimenCreated);
-            // Verify outcome
+            //arrange
+            var requests = new[] {new object(), new object(), new object()};
+            var requestQueue = new Queue<object>(requests);
+            var firstRequest = requestQueue.Dequeue();
+
+            var builder = new DelegatingSpecimenBuilder
+            {
+                OnCreate = (r, c) => requestQueue.Count > 0
+                    ? c.Resolve(requestQueue.Dequeue())
+                    : new object()
+            };
+
+            var sut = new TerminatingWithPathSpecimenBuilder(builder);
+            // Cause sut to be executed recursively for multiple times.
```

Comment:
Actually, yes :) However, such code was already present in other tests, so I thought that probably that's me, who needs a couple of minutes to understand why that works. Therefore, I've decided to leave a comment, so in future it will take less time to understand this nuance.

I'm not sure we can simplify that. Nothing tricky in fact happens here, rather it's a challenge for your attention 😉